src/js/base/ScrollList.tsx

```diff
@@ -15,6 +20,54 @@ const StyledScrollList = styled.div`
     z-index: 0;
 `;
 
+type ScrollListElementProps = {
+    className?: string;
+    fetchNextPage: (options?: FetchNextPageOptions) => Promise<InfiniteQueryObserverResult>;
+    isFetchingNextPage: boolean;
+    isLoading: boolean;
+    items: unknown[];
+    renderRow: (item: unknown) => void;
+};
+
+/**
+ * An infinitely scrolling list of items within an element
+ *
+ * @param className - The class name used for the scroll list
+ * @param fetchNextPage - A function which initiates fetching the next page
+ * @param isFetchingNextPage - Whether a new page is being fetched
+ * @param isLoading - Whether the first page is being fetched
+ * @param items - The list of items
+ * @param renderRow - A function which accepts an item and returns a react element
+ * @returns An infinitely scrolling list of items
+ */
+export function ScrollListElement({
+    className,
+    fetchNextPage,
+    isFetchingNextPage,
+    isLoading,
+    items,
+    renderRow,
+}: ScrollListElementProps) {
+    function onScroll() {
+        const scrollListElement = document.getElementById("scroll-list");
```

Comment:
```suggestion
    function onScroll(e) {
        const scrollListElement = e.target
```
or similar should work. Save having to rely on the id, which could quickly become a problem if more than one scrollList exists in a a single page